honeybee_grasshopper_energy/src/HB Schedule to Data.py

```diff
@@ -0,0 +1,116 @@
+        analysis_period_: An optional AnalysisPeriod to set the start and end datetimes
+            of the resulting DataCollection. The timestep of this AnalysisPeriod
+            will also be used to determine the timestep of the resulting DataCollection.
+        _week_start_day_: An optional text string to set the start day of the week
+            of the schedule timeseries data. Default: Sunday. Choose from the following:
+                * Sunday
+                * Monday
+                * Tuesday
+                * Wednesday
+                * Thursday
+                * Friday
+                * Saturday
+        holidays_: An optional list of DOYs (from 1 to 365) or a list of strings (eq: 25 Dec)
+            to represent the holidays in the resulting timeseries. Holiday schedules
+            will be used for these dates in the resulting timeseries.
+    
+    Returns:
+        data: A ladybug DataCollection representing the timeseries values of the schedule.
+            This can be used to visualize timeseries schedule values over the entire
+            period of a simulation using a component like the "LB Hourly Plot".
+"""
+
+ghenv.Component.Name = "HB Schedule to Data"
+ghenv.Component.NickName = 'SchToData'
+ghenv.Component.Message = '0.1.0'
+ghenv.Component.Category = "Energy"
+ghenv.Component.SubCategory = '2 :: Schedules'
+ghenv.Component.AdditionalHelpFromDocStrings = "2"
+
+
+try:  # import the honeybee-energy dependencies
+    from honeybee_energy.schedule.ruleset import ScheduleRuleset
+    from honeybee_energy.lib.schedules import schedule_by_name
+except ImportError as e:
+    raise ImportError('\nFailed to import honeybee_energy:\n\t{}'.format(e))
+try:  # import the ladybug dependencies
+    from ladybug.dt import Date, DateTime
+except ImportError as e:
+    raise ImportError('\nFailed to import ladybug:\n\t{}'.format(e))
+try:  # import ladybug_rhino dependencies
+    from ladybug_rhino.grasshopper import all_required_inputs
+except ImportError as e:
+    raise ImportError('\nFailed to import ladybug_rhino:\n\t{}'.format(e))
+
+
+if all_required_inputs(ghenv.Component):
+    # get the schedue from the library if it's a string
+    if isinstance(_schedule, str):
+        _schedule = schedule_by_name(_schedule)
+    
+    # process the _week_start_day_
+    week_start_day = 'Sunday' if _week_start_day_ is None else _week_start_day_.title()
+    
+    # process the analysis period if it is input
+    if analysis_period_ is not None:
+        start_date = analysis_period_.st_time.date()
+        end_date = analysis_period_.end_time.date()
+        timestep = analysis_period_.timestep
+    else:
+        start_date, end_date, timestep = Date(1, 1), Date(12, 31), 1
+    
+    # process the holidays_ if they are input
+    if len(holidays_) != 0 and holidays_[0] is not None:
+        holidays = []
+        for hol in holidays_:
+            if isinstance(hol, (int, str)):
```

Comment:
If this is something that you use in several components we may want to wrap this in a helper method like `try_get_date`.

Reply:
I realized that I can write this a lot more simply and align it with the other component that takes this type of input.  It's a bit less efficient but that really doesn't matter for something that gets used this infrequently.